obspy/io/hypodd/pha.py

```diff
@@ -111,19 +128,129 @@ def _read_pha(filename, inventory=None, id_map=None, id_default='.{}..{}',
+        idpha = idpha[:9]
+    while idpha == '' or idpha in used_ids:
+        idpha = str(counter[0])
+        counter[0] += 1
+    if idpha != evid:
+        eventid_map[evid] = idpha
+    used_ids.add(idpha)
+    return idpha, eventid_map
+
+
+def _write_pha(catalog, filename, eventid_map=None,
+               **kwargs):  # @UnusedVariable
+    """
+    Write a HypoDD PHA file.
+
+    .. warning::
+        This function should NOT be called directly, it registers via the
+        the :meth:`~obspy.core.event.Catalog.write` method of an
+        ObsPy :class:`~obspy.core.event.Catalog` object, call this instead.
+
+    :type catalog: :class:`~obspy.core.event.catalog.Catalog`
+    :param catalog: The ObsPy Catalog object to write.
+    :type filename: str or file
+    :param filename: Filename to write or open file-like object.
+    :param dict eventid_map: Desired mapping of event resource ids (dict keys)
+        to hypodd event ids (dict values).
+        HYPODD expects integer event ids with maximal 9 digits. If the event
+        resource id is not present in the mapping,
+        the event resource id is stripped of all non-digit characters and
+        truncated to a length of 9 chars. If this method does not generate a
+        valid hypodd event id, a counter starting at 1000 is used.
+
+    :returns: Dictionary eventid_map with mapping of event resource id to
+        hypodd event id. Items are only present if both ids are different.
+    """
+    lines = []
+    if eventid_map is None:
+        eventid_map = {}
+    args_map_eventid = (eventid_map, set(eventid_map.values()), [1000])
+    for event in catalog:
+        try:
+            ori = event.preferred_origin() or event.origins[0]
+        except IndexError:
+            warn('Skip writing events with missing origin')
+            continue
+        try:
+            mag = event.preferred_magnitude() or event.magnitudes[0]
+        except IndexError:
+            warn('Missing magnitude will be set to 0.0')
+            mag = 0.
+        else:
+            mag = mag.mag
+        evid = event.resource_id.id
+        evid = evid.split('/')[-1] if '/' in evid else evid
+        evid, eventid_map = _map_eventid(evid, *args_map_eventid)
+        rms = ori.quality.standard_error
+        rms = rms if rms is not None else 0.0
+        he1 = ori.latitude_errors.uncertainty
+        he2 = ori.longitude_errors.uncertainty
+        shortening = cos(deg2rad(ori.latitude))
+        he = max(0. if he1 is None else he1 * DEG2KM,
+                 0. if he2 is None else he2 * DEG2KM * shortening)
+        ve = ori.depth_errors.uncertainty
+        ve = 0. if ve is None else ve / 1000
+        line = PHA1.format(o=ori, depth=ori.depth / 1000, mag=mag,
+                           he=he, ve=ve, rms=rms,
+                           evid=evid)
+        lines.append(line)
+        weights = {}
+        for arrival in ori.arrivals:
+            weights[str(arrival.pick_id)] = arrival.time_weight
```

Comment:
```suggestion
            weights[str(arrival.pick_id)] = arrival.time_weight or 1.0
```
`ph2dt` complains if time_weight is `None`.